quickwit/quickwit-control-plane/src/ingest/ingest_controller.rs

```diff
@@ -706,9 +780,189 @@ impl IngestController {
         }
     }
 
-    pub fn rebalance_shards(&mut self) {
-        // TODO: As of now, it is only used for unit testing.
+    pub(crate) async fn rebalance_shards(
+        &mut self,
+        model: &mut ControlPlaneModel,
+        mailbox: &Mailbox<ControlPlane>,
+        progress: &Progress,
+    ) {
+        let Ok(_rebalance_permit) = self.rebalance_permits.clone().try_acquire_owned() else {
```

Comment:
```suggestion
        let Ok(rebalance_permit) = self.rebalance_permits.clone().try_acquire_owned() else {
```

`_blabla` is  for stuff that is unused. Here it is not unused... Of course, rust 1.77 is a pain and forces you to have put _ on unused struct stuff but still.